llvm/lib/Transforms/IPO/RepoPruning.cpp

```diff
@@ -123,14 +123,125 @@ repodefinition::DigestType toDigestType(pstore::index::digest D) {
+  X(rel_ro)                                                                    \
+  X(mergeable_1_byte_c_string)                                                 \
+  X(mergeable_2_byte_c_string)                                                 \
+  X(mergeable_4_byte_c_string)                                                 \
+  X(mergeable_const_4)                                                         \
+  X(mergeable_const_8)                                                         \
+  X(mergeable_const_16)                                                        \
+  X(mergeable_const_32)                                                        \
+  X(read_only)                                                                 \
+  X(thread_data)
+
+static void collectFragmentXFixupNames(
+    const pstore::database &Repository,
+    std::shared_ptr<const pstore::repo::fragment> const &Fragment,
+    StringSet<> &XFixupNames) {
+  for (const pstore::repo::section_kind Kind : *Fragment) {
+    assert(Fragment->has_section(Kind));
+
+#define X(k)                                                                   \
+  case pstore::repo::section_kind::k: {                                        \
+    auto Section = Fragment->atp<pstore::repo::section_kind::k>();             \
+    collectSectionXFixupName(Repository, *Section, XFixupNames);               \
+    break;                                                                     \
+  }
+
+    switch (Kind) {
+      PSTORE_GENERIC_SECTION_KINDS
+    case pstore::repo::section_kind::debug_line:
+    case pstore::repo::section_kind::dependent:
+    case pstore::repo::section_kind::bss:
+    case pstore::repo::section_kind::thread_bss:
+    case pstore::repo::section_kind::debug_string:
+    case pstore::repo::section_kind::debug_ranges:
+    case pstore::repo::section_kind::last:
+    case pstore::repo::section_kind::interp:
+      break;
+    }
+#undef X
+  }
+}
+
+// Reurn the GO's status as a tuple<InRepository, InModeuleFragments, Key>:
+// InRepository: return true if the GO is in the existing repository.
+// InModeuleFragments: reurn true if the GO is in the new module fragments.
+// Key: return GO's digest.
+static std::tuple<bool, bool, pstore::index::digest> hasExistingGO(
+    const pstore::database &Repository,
+    std::shared_ptr<const pstore::index::fragment_index> const &Fragments,
+    std::map<pstore::index::digest, const GlobalObject *> &ModuleFragments,
+    GlobalObject &GO, StringSet<> &XFixupNames,
+    DenseSet<GlobalObject *> &PrunedDiscardableIfUnusedGos) {
+  auto const Result = repodefinition::get(&GO);
+  assert(!Result.second && "The repo_definitio metadata should be created by "
+                           "the RepoMetadataGeneration pass!");
+  auto const Digest =
+      pstore::index::digest{Result.first.high(), Result.first.low()};
+
+  // Reurn true if the GO is in the existing repository.
```

Comment:
s/Reurn/Return/

This comment is somewhat misleading because the code does more than just return a value.

Reply:
Fixed, thanks.